packages/ember/tests/routing/decoupled_basic_test.js

```diff
@@ -0,0 +1,329 @@
+    });
+  }
+
+  [`@test An alternate template will pull in an alternate controller instead of controllername`](assert) {
+    this.add('route:home', Route.extend({
+      controllerName: 'foo',
+      renderTemplate() {
+        this.render('homepage');
+      }
+    }));
+    this.add('controller:foo', Controller.extend({
+      model: {
+        home: 'Comes from foo'
+      }
+    }));
+    this.add('controller:homepage', Controller.extend({
+      model: {
+        home: 'Comes from homepage'
+      }
+    }));
+
+    return this.visit('/').then(() => {
+      let text = this.$('p').text();
+
+      assert.equal(text, 'Comes from homepage',
+        'the homepage template was rendered'
+      );
+    });
+  }
+
+  [`@test The template will pull in an alternate controller via key/value`](assert) {
+    this.router.map(function() {
+      this.route('homepage', { path: '/' });
+    });
+
+    this.add('route:homepage', Route.extend({
+      renderTemplate() {
+        this.render({ controller: 'home' });
+      }
+    }));
+    this.add('controller:home', Controller.extend({
+      model: {
+        home: 'Comes from home.'
+      }
+    }));
+
+    return this.visit('/').then(() => {
+      let text = this.$('p').text();
+
+      assert.equal(text, 'Comes from home.',
+        'the homepage template was rendered from data from the HomeController'
+      );
+    });
+  }
+
+  [`@test The Homepage with explicit template name in renderTemplate and controller`](assert) {
+    this.add('controller:home', Controller.extend({
+      model: {
+        home: 'YES I AM HOME'
+      }
+    }));
+    this.add('route:home', Route.extend({
+      renderTemplate() {
+        this.render('homepage');
+      }
+    }));
+
+    return this.visit('/').then(() => {
+      let text = this.$('p').text();
+
+      assert.equal(text, 'YES I AM HOME',
+        'The homepage template was rendered'
+      );
+    });
+  }
+
+  [`@test Model passed via renderTemplate model is set as controller's model`](assert) {
+    this.addTemplate('bio', '<p>{{model.name}}</p>');
+    this.add('route:home', Route.extend({
+      renderTemplate() {
+        this.render('bio', {
+          model: { name: 'emberjs' }
+        });
+      }
+    }));
+
+    return this.visit('/').then(() => {
+      let text = this.$('p').text();
+
+      assert.equal(text, 'emberjs',
+        `Passed model was set as controller's model`
+      );
+    });
+  }
+
+  ['@test render uses templateName from route'](assert) {
+    this.addTemplate('the_real_home_template',
+      '<p>THIS IS THE REAL HOME</p>'
+    );
+    this.add('route:home', Route.extend({
+      templateName: 'the_real_home_template'
+    }));
+
+    return this.visit('/').then(() => {
+      let text = this.$('p').text();
+
+      assert.equal(text, 'THIS IS THE REAL HOME',
+        'the homepage template was rendered'
+      );
+    });
+  }
+
+  ['@test defining templateName allows other templates to be rendered'](assert) {
```

Comment:
@mixonic sorry, was currently adding more stuff to the file while you started the PR review. There are 5 more tests added at the end. Should be non-controversial edits.